rootclock.c

```diff
@@ -66,27 +87,67 @@ static void render_all(Drw *drw, Fnt *tf, Fnt *df, int show_date_flag,
                        Clr *bg_scm, Clr *time_scm, Clr *date_scm,
                        const char *time_fmt_s, const char *date_fmt_s,
                        int block_y_off_s, int line_spacing_s) {
+  char tbuf[64], dbuf[128];
   time_t now = time(NULL);
+  
+  if (now == (time_t)-1) {
+    /* time() failed, use fallback */
+    strcpy(tbuf, "--:--");
+    if (show_date_flag)
+      strcpy(dbuf, "Unknown Date");
+    goto skip_time_formatting;
```

Comment:
Avoid using `goto`, when both `time()` and `localtime()` fail, it would make more sense to quit the program with an error message to stderr rather than continue running broken.